### PR TITLE
ajd chapter 7

### PR DESCRIPTION
**Pull request by alexdeas**
Chapter 7

### CODE REVIEW

File: chapter07/ajd/exercise1.pl [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+my $target_dow = 1;        #  Monday is 1 ... Sunday is 7
+my @starting_directories = (".");
+
+# Using DateTime as it handles DST and is tidier
+my $date = DateTime->today();
+
+# Find the most recent Monday
+while ( $date->day_of_week != $target_dow ) {
+  $date->subtract( days => 1 );
+}
```
Comment (dickyince) [inline chapter07/ajd/exercise1.pl:22]: I can't see where this code was reviewed, not signed!

File: chapter07/ajd/exercise1.pl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  my $gather = sub {
+    my $mtime = ( stat $_ )[9];
+    if ( $mtime > $start
+      && $mtime < $stop ) {
+      push ( @files, File::Spec->canonpath( $File::Find::name ) );
+    }
+  };
+
+  my $yield = sub {
+    @files = File::Spec->no_upwards( @files ) ;
```
Comment (dickyince) [inline chapter07/ajd/exercise1.pl:50]: looks like someone has been reading CPAN

Comment (dickyince) [inline chapter07/ajd/exercise1.pl:50]: or indeed the chapter...

File: chapter07/ajd/exercise1.pl (hunk 1 above)
Comment (alexdeas) [inline chapter07/ajd/exercise1.pl:22]: You just reviewed it.

File: chapter07/ajd/exercise1.pl [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return wantarray ? @files : [ @files ];
```
Comment (thruddle) [inline chapter07/ajd/exercise1.pl:51]: Is there a reason you're populating an anonymous array ref instead of just using `\@files`?

Comment (utopia-2) [inline chapter07/ajd/exercise1.pl:51]: There's a _very_ good reason: clients of the closure can't mutate the captured lexical variables.

Comment (thruddle) [inline chapter07/ajd/exercise1.pl:51]: Fair enough then.

File: chapter07/ajd/exercise1.pl [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+use utf8;
+
+use local::lib; # Had to install DateTime locally
+
+use File::Find;
+use File::Spec;
+use DateTime;
+
+my $target_dow = 1;        #  Monday is 1 ... Sunday is 7
+my @starting_directories = (".");
```
Comment (Spinodal) [inline chapter07/ajd/exercise1.pl:14]: `my @starting_directories =`:open_mouth:`;`

**Pull request merged**